oncall/autoIssueAssign.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+autoAssign() {
+  echo "Current oncall is $1"
+  response=`curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $2" https://api.github.com/repos/microsoft/Oryx/issues | jq '.[]| select(.assignee == null or .assignee == "")' | jq '.number'`
+    arr=($response)
+    for i in "${arr[@]}";
+    do
+      curl -X POST "https://api.github.com/repos/microsoft/Oryx/issues/$i/assignees" -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $2" -d "{\"assignees\": ["\"$1\""]}"
+    done
+}
+
+file=$1
+token=$2
+
+if [ ! -f "$file" ]; then
+  echo "Error: Custom rotation file not found"
+  exit 1
+fi
+
+today=$(date +%s)
+
+while read line || [ -n "$line" ]; do
+  if [ "${line:0:1}" = "#" ]
+    then
+      continue
+  else
+    # Extract the name, start date, and end date from the line
+    name=$(echo $line | cut -d ' ' -f1)
+    startDate=$(echo $line | cut -d ' ' -f 2)
+    startDate=$(date -d $startDate +%s)
+    endDate=$(echo $line | cut -d ' ' -f 3)
+    endDate=$(date -d $endDate +%s)
+    # Check if the current date is within the date range
+    if [ "$today" -ge "$startDate" ] && [ "$today" -le "$endDate" ]; then
+      echo "Custom rotation found"
+      autoAssign $name $token
+      exit 0
+    fi
+  fi
+done < "$file"
+
+oncallList=("waliMSFT" "cormacpayne" "harryli0108" "snehapar9" "pauld-msft" "william-msft")
```

Comment:
Similar comment to the one left below for `customRotation.txt` -- I think it would be great if we could have the `oncallList` and `anchorDate` as repository variables that are able to be modified at any instance in case the rotation drastically changes and/or we need to add/remove people. I just want to minimize the number of PRs we need to make to support this automated workflow moving forward.